editor/scene_tree_editor.cpp

```diff
@@ -645,6 +645,13 @@ void SceneTreeEditor::_rename_node(ObjectID p_node, const String &p_name) {
 	TreeItem *item = _find(tree->get_root(), n->get_path());
 	ERR_FAIL_COND(!item);
 
+	// Nodes must have a non empty name, so resets it to previous text and show warning
+	if (p_name.empty()) {
+		item->set_text(0, n->get_name());
+		EditorNode::get_singleton()->show_warning(TTR("No name provided"));
```

Comment:
Maybe be a bit more explicit? "Empty node names are not allowed."

Reply:
Thanks for the suggestion. :)

Reply:
"Node name must have at least 1 visible letter." ?

Reply:
"Empty node names are not allowed." sounds better I think, you also include any character that's not 0 width this way.